src/key.rs

```diff
@@ -589,6 +627,25 @@ impl PublicKey {
             }
         }
     }
+
+    /// Returns the [`XOnlyPublicKey`] (and it's [`Parity`]) for this [`PublicKey`].
+    #[inline]
+    pub fn x_only_public_key<C: Signing>(&self, secp: &Secp256k1<C>) -> Result<(XOnlyPublicKey, Parity), Error> {
+        let mut pk_parity = 0;
+        unsafe {
+            let mut xonly_pk = ffi::XOnlyPublicKey::new();
+            let ret = ffi::secp256k1_xonly_pubkey_from_pubkey(
+                secp.ctx,
```

Comment:
Cool, cheers. How did you know this? Am I right in thinking its because `ffi::secp256k1_context_no_precomp` is used when the ffi function has a secp paramater for API uniformity but its not needed, 'needed' being only when we have to do a multiplication by `G` i.e., only when going sk -> pk? Am I understanding this correctly?

Reply:
> How did you know this?

Part of it is experience. But you can always refer to underlying secp256k1 documentation. If it requires no context, it documentation states "ctx: pointer to a context object". But if some specific context is needed, it always say " pointer to context initialized for verification" . 

> m I right in thinking its because ffi::secp256k1_context_no_precomp is used when the ffi function has a secp paramater for API uniformity but its not needed

Yes, exactly. In the future, this parameter might be used to offer CPU architecture optimizations, but currently, it is not used. 

> 'needed' being only when we have to do a multiplication by G i.e., only when going sk -> pk? 

Yes, correct.